util/stream.go

```diff
@@ -64,9 +67,8 @@ func NewMessageStream(conn net.Conn, parser Parser) *MessageStream {
 	go m.outbound()
 	go m.inbound()
 
-	for i := 0; i < 25; i++ {
```

Comment:
@unclejack This will need some good amount of (stress)testing. In this path, we are less concerned about the order of messages. I would suggest leaving the parallelism in place for now and just addressing the race condition and the leak.